packages/discord.js/src/structures/MentionableSelectMenuInteraction.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const MessageComponentInteraction = require('./MessageComponentInteraction');
+const Events = require('../util/Events');
+
+/**
+ * Represents a {@link ComponentType.MentionableSelect} select menu interaction.
+ * @extends {MessageComponentInteraction}
+ */
+class MentionableSelectMenuInteraction extends MessageComponentInteraction {
+  constructor(client, data) {
+    super(client, data);
+
+    const { members, users, roles } = data.data.resolved || {};
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, User>}
+     */
+    this.users = new Collection();
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, GuildMember|APIGuildMember>}
+     */
+    this.members = new Collection();
+
+    /**
+     * Collection of the selected roles
+     * @type {Collection<Snowflake, Role|APIRole>}
+     */
+    this.roles = new Collection();
+
+    if (members) {
+      for (const [id, member] of Object.entries(members)) {
+        const user = users[id];
+        if (!user) {
+          this.client.emit(
+            Events.Debug,
+            `[MentionableSelectMenuInteraction] Received a member without a user, skipping ${id}`,
```

Comment:
Should we skip it? Should we add it with user null? @discordjs/core

Reply:
Skipping should be ok, it's actually theoretically impossible to not have the user in `users` - this condition is here just as a last resort sanity check to prevent a hard crash.

Also, from a bug I encountered earlier where I wasn't passing in the user at all, the manager was straight up throwing without it - it relies on `member.user.id` (and potentially some other things).